vm/vm.go

```diff
@@ -1209,6 +1212,30 @@ func (vm *VM) restoreAcceptedQueue(ctx context.Context) error {
 	return nil
 }
 
+func (*VM) removeSuffixes(values [][]byte, errors []error) ([][]byte, []error) {
+	returnValues := make([][]byte, 0, len(values))
+	returnErrors := make([]error, 0, len(values))
+
+	// Invariant: len(values) == len(errors)
+	for i, v := range values {
+		if errors[i] != nil {
+			returnErrors = append(returnErrors, errors[i])
+			returnValues = append(returnValues, nil)
+			continue
+		}
+		// Time to unsuffix
+		if len(v) < consts.Uint64Len {
+			returnErrors = append(returnErrors, ErrValueTooShortForPrefix)
+			returnValues = append(returnValues, nil)
+			continue
+		}
+		returnValues = append(returnValues, v[:len(v)-consts.Uint64Len])
+		returnErrors = append(returnErrors, nil)
+	}
+
+	return returnValues, returnErrors
+}
+
```

Comment:
Does this need to be coupled to the VM? In effect, we are grabbing the state from the VM's merkledb, could we apply the same transformation we normally do to a merkledb view before giving actions read access to it?

This function gives raw access to the statedb, should we leave to the caller (that knows where the information is used) to strip the suffixes / transform the data as needed?